server/src/main/java/com/epam/aidial/core/server/ContextManager.java

```
@@ -0,0 +1,71 @@
+package com.epam.aidial.core.server;
+
+import io.vertx.core.Context;
+import io.vertx.core.Vertx;
+import lombok.experimental.UtilityClass;
+
+@UtilityClass
+public class ContextManager {
+
+    private static final String PROXY_CONTEXT_KEY = "proxyContext";
+
+    /**
+     * Set ProxyContext in Vertx context only.
+     * This simplifies context management by storing the entire ProxyContext object.
+     * The AutoEnrichedOtelJsonLayout will extract fields directly from ProxyContext.
+     */
+    public static void setProxyContext(ProxyContext proxyContext) {
+        if (proxyContext == null) {
+            return;
+        }
+
+        // Store only the ProxyContext object in Vertx context
+        Context vertxContext = Vertx.currentContext();
+        if (vertxContext != null) {
```

Should we throw an exception if the context is null?

---

No, i think we should not throw an exception when the context is null,
the current implementation correctly handles null contexts by returning early,
which is the appropriate behavior for several reasons:

1. Graceful degradation: The application can continue functioning even without context information.
  Logging and tracing are supplementary features that shouldn't crash the application.
2. Common scenario: Null contexts can occur in legitimate situations:
    - During startup/shutdown
    - Background tasks not associated with requests
    - Internal system operations
    - Test environments
3. Performance: Checking for null and returning early is more efficient than throwing and catching exceptions.

---

My believe when context is null is unlikely possible. We hide the fact that's something wrong. Throwing exception could reveal a problem proactively.

Ok. Let it be so.